src/network/ledger.rs

```diff
@@ -812,16 +812,35 @@ impl<N: Network, E: Environment> Ledger<N, E> {
                 return;
             }
 
+            // Filter out any pre-existing block requests for the peer.
+            let mut new_block_heights = Vec::new();
```

Comment:
Is it worth pre-allocating the vec here?

Reply:
We can't predict how many requests are new; unless the usual case is that all/most of them are guaranteed to be new, but I don't know that :smiley:.